controller/registries/consul/config.go

```diff
@@ -122,22 +128,24 @@ func (reg *Consul) Start(c registrytype.RegistryConfig) error {
 	}
 	go func() {
 		reg.logger.Infof("start refreshing services")
-		ticker := time.NewTicker(dur)
-		//q := consulapi.QueryOptions{
-		//    WaitTime: dur,
-		//}
-		defer ticker.Stop()
+		q := &consulapi.QueryOptions{
+			WaitTime: dur,
+		}
 		for {
 			select {
-			case <-ticker.C:
-				err := reg.refresh()
-				if err != nil {
-					reg.logger.Errorf("failed to refresh services, err: %v", err)
-				}
 			case <-reg.done:
 				reg.logger.Infof("stop refreshing services")
 				return
+
+			default:
+			}
+			services, meta, err := reg.client.consulCatalog.Services(q)
+			if err != nil {
+				reg.logger.Errorf("failed to get services, err: %v", err)
```

Comment:
Add a sleep and skip this turn would be better? So we can retry after some time.